misc/onboarding_command.py

```diff
@@ -0,0 +1,304 @@
+			</div>
+		</div>
+		<div class="step">
+			<div class="step-title">Step 6: Pieces Copilot</div>
+			<div class="step-description">
+				{copilot_status}
+			</div>
+		</div>
+		<div class="step">
+			<div class="step-title">Step 7: Explore the Features</div>
+			<div class="step-description">
+				Check out the <a href="https://example.com/docs">documentation</a> to learn about all the features and how to use them effectively.
+			</div>
+		</div>
+		<div class="step">
+			<div class="step-title">Step 8: Get Support</div>
+			<div class="step-description">
+				Feeling stuck or encountering bugs? No worries! Head over to your command palette and select <code><a href="subl:pieces_support">Pieces: Get Support</a></code>. You'll discover a treasure trove of useful resources to help you out!
+			</div>
+		</div>
+		<div class="step">
+			<div class="step-title">Step 9: Join the Community</div>
+			<div class="step-description">
+				Don't forget to join our <a href="https://docs.pieces.app/community/">community</a><br>
+				Don't forget also to check the <a href="subl:pieces_about">Pieces: About</a> command for your command palette.
+			</div>
+		</div>
+		<p>Enjoy using the Package! :)</p>
+		<a href="subl:pieces_onboarding">Reload the view</a>
+		<a href="subl:pieces_reset_onboarding">Rest Onboarding</a>
+	</div>
+</body>
+"""	
+def green(text:str) -> str:
+	return f'<span style="color:var(--greenish)">{text}</span>'
+
+def red(text:str) -> str:
+	return f'<span style="color:var(--redish)">{text}</span>'
+
+def subl_onboarding_commands(title,cmd):	
+	return f"""<a href='subl:{sublime.html_format_command("pieces_onboarding_commands",args={"cmd":cmd})}'>{title}</a>"""
+
+class PiecesOnboardingCommand(sublime_plugin.WindowCommand):
+	SHEET_NAME = "Welcome to Pieces 🎉🎉!"
+	calls = {}
+	clearing_time = time.time()
+	sheet_id = None
+
+	lazy_load_status = {
+		"pieces_os_status":"Checking Pieces OS",
+	} # Maps a function name to the status to be displayed when it is loading
+
+	ONBOARDING_SETTINGS_PATH = os.path.join(PiecesSettings.PIECES_USER_DIRECTORY, "onboarding_settings.json")
+	
+	def run(self):
+		sheet = sublime.HtmlSheet(self.sheet_id) if self.sheet_id in self.get_html_sheet_ids() else self.window.new_html_sheet(self.SHEET_NAME,"") 
+		self.sheet_id = sheet.id()
+		self.reload(sheet)
+
+	@staticmethod
+	def get_html_sheet_ids() -> List[int]:
+		ids = []
+		for window in sublime.windows():
+			for sheet in window.sheets():
+				if isinstance(sheet,sublime.HtmlSheet):
+					ids.append(sheet.id())
+		return ids
+
+	def reload(self,sheet):
+		kwargs = {
+			"os_status":self.pieces_os_status(),
+			"create_status":self.create_command_status(),
+			"open_status":self.open_asset_command_status(),
+			"search_status":self.search_command_status(),
+			"ask_status":self.ask_question_command_status(),
+			"copilot_status":self.copilot_status(),
+			"css":CSS,
+		}
+		sheet.set_contents(html_template.format(**kwargs))
+
+
+	def _lazy_load(func):
```

Comment:
why do we need this?

Reply:
Decorator to annotate the functions that will take time to load check pieces os takes sometimes a couple of seconds so we can show a loading status until the status is loaded